src/FS.Embedded.Manifest.Task/Manifest.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Extensions.FileProviders.Embedded.Manifest.Task.Internal;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest.Task
+{
+    public class Manifest
+    {
+        public Entry Root { get; set; } = Entry.Directory("");
+
+        public void AddElement(string originalPath, string assemblyResourceName)
+        {
+            var paths = originalPath.Split(Path.DirectorySeparatorChar);
```

Comment:
You're missing argument validation here. originalPath can be null and that'll result in NullReferenceException